### PR TITLE
Retrieve the email address from Twitter.

### PR DESCRIPTION
**Pull request by Tratcher**
#765 @HaoK @muratg 

And some cleanup of the twitter handler and sample code.

### CODE REVIEW

File: src/Microsoft.AspNetCore.Authentication.Twitter/TwitterHandler.cs [hunk 1]
```diff
@@ -288,6 +296,76 @@ public TwitterHandler(HttpClient httpClient)
...
+            var request = new HttpRequestMessage(HttpMethod.Get, resource_url + "?include_email=true");
+            request.Headers.Add("Authorization", authorizationHeaderBuilder.ToString());
+
+            var response = await _httpClient.SendAsync(request, Context.RequestAborted);
+            if (!response.IsSuccessStatusCode)
+            {
+                Logger.LogError("Email request failed with a status code of " + response.StatusCode);
+                response.EnsureSuccessStatusCode(); // throw
+            }
+            string responseText = await response.Content.ReadAsStringAsync();
```
Comment (muratg) [inline src/Microsoft.AspNetCore.Authentication.Twitter/TwitterHandler.cs]: nit: var

File: src/Microsoft.AspNetCore.Authentication.Twitter/TwitterOptions.cs [hunk 2]
```diff
@@ -39,6 +39,14 @@ public TwitterOptions()
         public string ConsumerSecret { get; set; }
 
         /// <summary>
+        /// Requests the middleware to retrieve the users detials, including e-mail address, durring the
```
Comment (muratg) [inline src/Microsoft.AspNetCore.Authentication.Twitter/TwitterOptions.cs]: tpyo: detials, durring

File: src/Microsoft.AspNetCore.Authentication.Twitter/TwitterHandler.cs [hunk 3]
```diff
@@ -288,6 +296,76 @@ public TwitterHandler(HttpClient httpClient)
             };
         }
 
+        // https://dev.twitter.com/rest/reference/get/account/verify_credentials
+        private async Task<JObject> RetrieveUserDetailsAsync(AccessToken accessToken, ClaimsIdentity identity)
+        {
```
Comment (muratg) [inline src/Microsoft.AspNetCore.Authentication.Twitter/TwitterHandler.cs]: Do we need null checks here?

**Comment by muratg** [comment]
A few comments, otherwise :shipit:

File: src/Microsoft.AspNetCore.Authentication.Twitter/TwitterHandler.cs (hunk 3 above)
Comment (Tratcher) [inline src/Microsoft.AspNetCore.Authentication.Twitter/TwitterHandler.cs]: Not usually for private APIs.

**Comment by oliverjanik** [comment]
How do I use this?

**Comment by Tratcher** [comment]
See https://github.com/aspnet/Security/issues/765#issue-145708666

**Pull request merged**